### ISSUE TITLE
a question about self.vocab

### ISSUE THREAD

**Issue by tomtang110** [benywon/ChineseBert#3]
![2019-01-17 18 13 06](https://user-images.githubusercontent.com/33299522/51311325-a141e900-1a83-11e9-9488-23c427f7ddf7.png)
Could you explain why you add self.vocab_size between question id and answer id?

**Comment by tomtang110** [comment]
想问下，你们训练的单词只是针对你们的word2id.obj文件吗？ 如果我自己建一套我自己的word2id可以使用你们的模型吗？

**Comment by benywon** [comment]
Definitely!! Different word2id would project the same word to a different id. So you should use my word2id.obj. BTW, 57777 words is not very small as we use the sentencepiece word tokenizer, so OOV is not a problem.

**Comment by tomtang110** [comment]
But, I need more than 450000 words, 57777 for 450000 is few. It is so upset. Therefore, for most companies, I think Bert is still difficult for training, even, fine-tune.

**Comment by benywon** [comment]
Oh, that so bad, if you have your own vocab, this application may not suitable for you. Nevertheless, you can use my codes to train your own BERT.

**Comment by tomtang110** [comment]
haha, But my company has no so affluent hardware equipment. My boss told me, next year, they would introduce cloud server, But at that time, I have finished my internship. Actually, I have made a machine comprehension reading system, I used the QAnet based on transformer as model, so I would like to try to use bert to train the Dureader dataset. However, it seems its value is too high to train it.
